packages/jaeger-ui/src/utils/date.tsx

```diff
@@ -40,67 +43,73 @@ const UNIT_STEPS: { unit: string; microseconds: number; ofPrevious: number }[] =
   { unit: 'μs', microseconds: 1, ofPrevious: 1000 },
 ];
 
-const timeUnitToShortTermMapper = {
+type ShortTimeUnit = 'μs' | 'ms' | 's' | 'm' | 'h' | 'd';
+type LongTimeUnit = 'microseconds' | 'milliseconds' | 'seconds' | 'minutes' | 'hours' | 'days';
+
+const timeUnitToShortTermMapper: {
+  [key in LongTimeUnit]: ShortTimeUnit;
+} = {
+  microseconds: 'μs',
   milliseconds: 'ms',
   seconds: 's',
   minutes: 'm',
   hours: 'h',
   days: 'd',
-};
+} as const;
 
 /**
  * @param {number} timestamp
  * @param {number} initialTimestamp
  * @param {number} totalDuration
  * @return {number} 0-100 percentage
  */
-export function getPercentageOfDuration(duration: number, totalDuration: number) {
+export function getPercentageOfDuration(duration: number, totalDuration: number): number {
   return (duration / totalDuration) * 100;
 }
 
-const quantizeDuration = (duration: number, floatPrecision: number, conversionFactor: number) =>
+const quantizeDuration = (duration: number, floatPrecision: number, conversionFactor: number): number =>
   toFloatPrecision(duration / conversionFactor, floatPrecision) * conversionFactor;
 
 /**
  * @param {number} duration (in microseconds)
  * @return {string} formatted, unit-labelled string with time in milliseconds
  */
-export function formatDate(duration: number) {
-  return moment(duration / ONE_MILLISECOND).format(STANDARD_DATE_FORMAT);
+export function formatDate(duration: number): string {
+  return dayjs(duration / ONE_MILLISECOND).format(STANDARD_DATE_FORMAT);
 }
 
 /**
  * @param {number} duration (in microseconds)
  * @return {string} formatted, unit-labelled string with time in milliseconds
  */
-export function formatTime(duration: number) {
-  return moment(duration / ONE_MILLISECOND).format(STANDARD_TIME_FORMAT);
+export function formatTime(duration: number): string {
+  return dayjs(duration / ONE_MILLISECOND).format(STANDARD_TIME_FORMAT);
 }
 
 /**
  * @param {number} duration (in microseconds)
  * @return {string} formatted, unit-labelled string with time in milliseconds
  */
-export function formatDatetime(duration: number) {
-  return moment(duration / ONE_MILLISECOND).format(STANDARD_DATETIME_FORMAT);
+export function formatDatetime(duration: number): string {
+  return dayjs(duration / ONE_MILLISECOND).format(STANDARD_DATETIME_FORMAT);
 }
 
 /**
  * @param {number} duration (in microseconds)
  * @return {string} formatted, unit-labelled string with time in milliseconds
  */
-export function formatMillisecondTime(duration: number) {
+export function formatMillisecondTime(duration: number): string {
```

Comment:
Please add tests.

Side note: Many of these functions feel poorly conceived to me. Like `formatDate()` on L77 that takes `duration` as input - I don't really understand that that means, is it interpreting duration as offset from epoch to calculate the date and then format it? Comments could use improvement by showing examples.